ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/ParticipantTestContext.scala

```diff
@@ -286,11 +296,75 @@ private[testtool] final class ParticipantTestContext private[participant] (
   def transactionTrees(take: Int, parties: Party*): Future[Vector[TransactionTree]] =
     transactionTrees(take, getTransactionsRequest(parties))
 
+  /**
+    * Create a [[GetTransactionByIdRequest]] with an identifier and a set of [[Party]] objects.
+    * You should use this only when you need to tweak the request of [[transactionTreeById]] or
+    * [[flatTransactionById]], otherwise use the shortcut override that allows you to directly
+    * pass the identifier and parties.
+    */
+  def getTransactionByIdRequest(
+      transactionId: String,
+      parties: Seq[Party]): GetTransactionByIdRequest =
+    new GetTransactionByIdRequest(ledgerId, transactionId, Tag.unsubst(parties))
+
+  /**
+    * Non-managed version of [[transactionTreeById]], use this only if you need to tweak the request (i.e. to test low-level details)
+    */
+  def transactionTreeById(request: GetTransactionByIdRequest): Future[TransactionTree] =
```

Comment:
Do you want to move the "low-level" testing methods into a nested object?
Something like:
```
class ParticipantTestContext {
  object lowLevel {
    // low-level methods here
  }
```
This way it is fairly obvious that these are discouraged from general purpose use. What do you think?

Reply:
I was thinking of the same and I'm in favor of this (also to keep the "actual" API a bit leaner and cleaner). I would address this as part of a different ticket though. Is that OK with you?

Reply:
Sure, thanks 👍